main.py

```diff
@@ -110,61 +129,70 @@ def upload_file(
         f"as {object_name or file_name}"
     )
     # If S3 object_name was not specified, use file_name
-    if object_name is None:
+    if not object_name:
         object_name = file_name
 
     # Upload the file
-    s3_client = boto3.client('s3',
-                             region_name="us-east-1",
-                             aws_access_key_id=ACCESS_KEY,
-                             aws_secret_access_key=SECRET_KEY)
+    s3_client = boto3.client(
+        service_name="s3",
+        region_name="us-east-1",
+        aws_access_key_id=ACCESS_KEY,
+        aws_secret_access_key=SECRET_KEY
+    )
     try:
-        s3_client.upload_file(file_name, bucket, object_name,
-                              ExtraArgs={'ContentType': content_type})
+        s3_client.upload_file(
+            file_name,
+            bucket,
+            object_name,
+            ExtraArgs={"ContentType": content_type}
+        )
     except ClientError as e:
         logging.error(e)
         return False
     return True
 
 
 def upload_coverage_file(
-        coverage_percentage,
-        badge_name,
-        bucket
-):
+    coverage_percentage: int,
+    badge_name: str,
+    bucket: str
+) -> bool:
     """
     Upload a file with the coverage percentage to an S3 bucket.
 
     :param int coverage_percentage: Current coverage  percentage
     :param str badge_name: Name of the badge file
     :param str bucket: Bucket to upload to
-    :param str|None object_name: S3 object name. If not specified then
-        file_name is used
     :return: True if file was uploaded, else False
     :rtype: bool
     """
     coverage_file = os.path.splitext(badge_name)[0] + ".txt"
-    temp_coverage_file = "coverage-percentage.tx"
+    temp_coverage_file = "coverage-percentage.txt"
     logging.info(
-        f"Writing coverage percentage to file {coverage_file} to {bucket} ."
+        f"Writing coverage percentage to file {coverage_file} to {bucket}."
     )
-    with open(temp_coverage_file, 'w') as f:
-        logging.info(f"Coverage  : {coverage_percentage}")
+    with open(temp_coverage_file, "w") as f:
+        logging.info(f"Coverage: {coverage_percentage}")
         f.write(str(coverage_percentage))
 
     # Upload the file
-    upload_file(temp_coverage_file, bucket, "text/plain", coverage_file)
+    return upload_file(
+        file_name=temp_coverage_file,
```

Comment:
We should check with @TDaglis  in order to also fix the issue with the cached images. @TDaglis  have you found a solution on this one?

Reply:
@anmarkoulis @TDaglis https://stackoverflow.com/a/48989966 maybe by adding the`CacheControl: "no-cache"` argument in ExtraArgs?

Reply:
Tom's answer is correct. If you have caching issues, this could help. But keep in mind that Github has also caching for their images in their pages (I do not remember if they have for hotlinking images though), so this might not be fixed without doing `Ctrl + R`.  But that would flush the cache for Github which would show the un-cached image of S3

Reply:
@elektracodes Yes, exactly. Most probably Github caches as well, because when I run it locally and check the url, when I reload I get the new badge.

Reply:
Github's Documentation: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-anonymized-urls#an-image-that-changed-recently-is-not-updating